### ISSUE TITLE
Use switch expressions where appropriate

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#28180]
I don't think we make much use of switch statements, but we should see if there are places where using switch expressions will make the code more concise.

**Comment by philwebb** [comment]
I tried to use one today and got:

```
> Task :spring-boot-project:spring-boot-docs:aggregatedJavadoc FAILED
/Volumes/Data/projects/spring-boot/code30x/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/AotProcessor.java:137: error: switch expressions are not supported in -source 8
                return switch (kind) {
                       ^
  (use -source 14 or higher to enable switch expressions)
/Volumes/Data/projects/spring-boot/code30x/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/AotProcessor.java:138: error: switch rules are not supported in -source 8
                case SOURCE -> this.sourceOutput;
                            ^
  (use -source 14 or higher to enable switch rules)
2 errors

```

**Comment by wilkinsona** [comment]
@philwebb I think I've fixed that in https://github.com/spring-projects/spring-boot/issues/31210. You should be able to use switch expressions on main now.

**Comment by dreis2211** [comment]
I think there needs some work to be done on the formatting end:

Running `:format` turns this:
```
return switch (getErrorProperties().getIncludeBindingErrors()) {
     case ALWAYS -> true;
     case ON_PARAM -> getErrorsParameter(request);
     default -> false;
};
```
into this:
```
return switch (getErrorProperties().getIncludeBindingErrors()) {
case ALWAYS -> true;
case ON_PARAM -> getErrorsParameter(request);
default -> false;
};
```

Arguably, the missing indentation makes it harder to read.

Otherwise this is ready on my local branch.

**Comment by wilkinsona** [comment]
Thanks, @dreis2211. I prefer the pre-formatted version too. We can change this in Spring Java Format with this setting:

```
org.eclipse.jdt.core.formatter.indent_switchstatements_compare_to_switch=true
```

It also affects a "traditional" switch. This:

```
protected boolean isIncludeBindingErrors(HttpServletRequest request, MediaType produces) {
    switch (getErrorProperties().getIncludeBindingErrors()) {
    case ALWAYS:
        return true;
    case ON_PARAM:
        return getErrorsParameter(request);
    default:
        return false;
    }
}
```

Will become this:

```
protected boolean isIncludeBindingErrors(HttpServletRequest request, MediaType produces) {
    switch (getErrorProperties().getIncludeBindingErrors()) {
        case ALWAYS:
            return true;
        case ON_PARAM:
            return getErrorsParameter(request);
        default:
            return false;
    }
}
```

**Comment by dreis2211** [comment]
@wilkinsona thanks.

I will hold my changes back until this is done. Could we maybe assign this issue to me, so nobody starts to work on it in parallel?

**Comment by dreis2211** [comment]
There you go #31527 @wilkinsona

**Comment by wilkinsona** [comment]
Thanks, Christoph. Closing in favor of #31527.
